src/main.cc

```diff
@@ -908,9 +914,8 @@ int run_filter(StringView keystr, ConstArrayView<StringView> files, bool quiet,
             }
             catch (runtime_error& err)
             {
-                if (not quiet)
-                    write_stderr(format("error while applying keys to buffer '{}': {}\n",
-                                        buffer.display_name(), err.what()));
+                throw filter_key_error{format("could not apply keys to buffer '{}': {}",
+                                              buffer.display_name(), err.what())};
```

Comment:
When given multiple files, this will stop all processing at the first failing one, which I dont think would be the desired behaviour.

Reply:
I did this on purpose, if errors do not interrupt the processing of the subsequent files, then there's no need to have an exit code - there will be no errors, only warnings.

Users who don't care can wrap the macro in `:try "…"<ret>`, or even spawn `kak -f` on each individual file (ignoring the non-zero exit code).

Users who expect their macro to apply will know immediately something is off, it seems inconvenient to keep executing keys further when there has already been an error.

Reply:
It seems that using `kak -f 'sfoo<ret>cbar<esc>' directory/*` is a reasonable use case for replacing all foo with bar in files appearing in the directory.

With this logic, you would end up with a set of file already replaced, and all the files following the first one not containing foo not replaced. A sane behaviour would be to either apply to all files, or none, but not an unpredicitible partial set.

Reply:
What about the cases whereby users expect their macro to apply perfectly to *all* the files passed as argument, or when the string of primitives errors out in the middle of it, and not at the very beginning?

Why would the editor apply what it knows is a broken macro to all the files, leaving them *all* in a broken state?

We can't assume *all errors* can safely be ignored because of the one case that would allow `-f` to be used as a drop-in replacement for `sed`.

Reply:
Regarding `sed` I believe it will just apply to all files, not complaining about files that do not have any match.

Also, in my view, `sfoo<ret>cbar<esc>` on a file that does not contain any foo does not fail, it mearly stops early because it got to an empty selection set. It still did exactly what was asked for, replacing all `foo` with `bar`, wether the user expects all files to contain foo or not is something Kakoune can not guess.